components/mpas-ocean/driver/ECDA_mpaso_obs_mod.F

```diff
@@ -0,0 +1,636 @@
+                                  shr_file_freeUnit,    &
+                                  shr_file_setIO,       &
+                                  shr_file_getLogUnit,  &
+                                  shr_file_setLogUnit,  &
+                                  shr_file_getLogLevel, &
+                                  shr_file_setLogLevel
+
+  implicit none
+  public :: ocn_obs_init, set_obs_type_value, ocn_obs_type_register, &
+            observation_deallocate, phys2ij_mercator
+  
+  integer, parameter, public :: max_OCN_obs_type =  10    ! Currently using T,S,U,V,zeta 
+  
+  integer, parameter, public :: O_rea_TEMP_ID = 1     ! index of reanalysis ocn temperature
+  integer, parameter, public :: O_rea_SALT_ID = 2     ! index of reanalysis ocn salinity
+  integer, parameter, public :: O_rea_VEL_ID = 3     ! index of reanalysis ocn velocity
+  integer, parameter, public :: O_rea_SSH_ID = 4     ! index of reanalysis ocn surface height
+
+
+  integer, parameter, public :: O_GRID_MODEL     = 1        ! instrument type for reanalysis with the same grids as ocean model 
+  integer, parameter, public :: O_GRID_NONMODEL =  2         ! grid observation but not the model grid
+  integer, parameter, public :: O_profile =  3         ! grid observation but not the model grid
+
+  integer, public :: NUM_OCN_obs_type
+!  type(ECDA_obs_type),public         ::  OCN_obs_data(max_OCN_obs_type) 
+  type(ECDA_obs_type),allocatable,public         ::  OCN_obs_data(:) 
+
+contains
+
+!==============================================================================
+!BOP
+!   !ROUTINE: ocn_obs_init
+!   initialize the OCN obs for DA
+!   define the observations
+!   !INTERFACE:
+
+    subroutine ocn_obs_init()
+!EOP
+
+!BOC
+        implicit none
+        ! --- Local variables ---
+        character(*), parameter :: subName =   '(ocn_obs_init) '
+        integer :: iob,i,j,k,ierr
+        integer :: inst_type
+!        logical :: update_states(n_OCN_state),required_states(n_OCN_state)
+!        allocate (OCN_obs_data(max_OCN_obs_type)        
+        do iob=1,NUM_OCN_obs_type
+           OCN_obs_data(iob)%assimilate=.False.    
+           allocate (OCN_obs_data(iob)%required_states(n_OCN_state), stat=ierr)
+           allocate (OCN_obs_data(iob)%update_states(n_OCN_state), stat=ierr)
+           OCN_obs_data(iob)%required_states = .FALSE.
+           OCN_obs_data(iob)%update_states   = .FALSE.
+           OCN_obs_data(iob)%grid_type    = O_GRID_MODEL       ! set model grid as default
+        enddo
+
+     end subroutine ocn_obs_init
+    
+!==============================================================================
+!BOP
+! ! Rountine set_obs_type_value(obs_data,...)
+! ! set the obs configures
+ 
+    subroutine set_obs_type_value(ID,obs_name,filename,dm_name,inst_type,assimilate, required_states, update_states)
+      
+! EOP 
+       integer, intent(in)                                    ::      ID
```

Comment:
can you add brief comments of what these variables mean?  especially shorter ones like 'dm_name' and 'inst_type'

Reply:
I will change them with long names and add comments